docs/src/main/asciidoc/security-jdbc-guide.adoc

```diff
@@ -0,0 +1,318 @@
+
+```bash
+$ curl -i -X GET http://localhost:8080/api/public
+HTTP/1.1 200 OK
+Content-Length: 6
+Content-Type: text/plain;charset=UTF-8
+
+public%
+```
+
+Now, let's try a to hit a protected resources anonymously.
+
+```bash
+$ curl -i -X GET http://localhost:8080/api/admin
+HTTP/1.1 401 Unauthorized
+Content-Length: 14
+Content-Type: text/html;charset=UTF-8
+
+Not authorized%
+```
+
+So far, so good,  now let's try with an allowed user.
+
+```bash
+$ curl -i -X GET -u admin:admin http://localhost:8080/api/admin
+HTTP/1.1 200 OK
+Content-Length: 5
+Content-Type: text/plain;charset=UTF-8
+
+admin%
+```
+By providing the `admin:admin` credentials, the extension succeed to authenticate the user and load its roles.
+The `admin` user is authorized to access to the protected resources.
+
+The  user the `admin` should be forbid to access a resources protected with `@RolesAllowed("user")` because it hasn't this role.
+```bash
+$ curl -i -X GET -u admin:admin http://localhost:8080/api/users/me
+HTTP/1.1 403 Forbidden
+Content-Length: 34
+Content-Type: text/html;charset=UTF-8
+
+Access forbidden: role not allowed%
+```
+
+Finally, using the user `user` works and the security context contains the principal details (username for instance).
+```bash
+curl -i -X GET -u user:user http://localhost:8080/api/users/me
+HTTP/1.1 200 OK
+Content-Length: 4
+Content-Type: application/json
+
+user%
+```
+
+== Advanced Configuration
+
+This guide only covered an easy use case, the extension offers multiple datasources, multiple principal queries configuration as well as a bcrypt password mapper.
+
+[source,properties]
+--
+quarkus.datasource.url=jdbc:h2:mem:multiple-data-sources-users;DB_CLOSE_DELAY=-1;DB_CLOSE_DELAY=-1;INIT=RUNSCRIPT FROM 'classpath:multiple-data-sources/users.sql'
+quarkus.datasource.driver=org.h2.Driver
+quarkus.datasource.username=sa
+quarkus.datasource.password=sa
+
+quarkus.datasource.permissions.url=jdbc:h2:mem:multiple-data-sources-permissions;DB_CLOSE_DELAY=-1;DB_CLOSE_DELAY=-1;INIT=RUNSCRIPT FROM 'classpath:multiple-data-sources/permissions.sql'
+quarkus.datasource.permissions.driver=org.h2.Driver
+quarkus.datasource.permissions.username=sa
+quarkus.datasource.permissions.password=sa
+
+quarkus.security.jdbc.enabled=true
+quarkus.security.jdbc.principal-query.sql=SELECT u.password FROM test_user u WHERE u.username=?
+quarkus.security.jdbc.principal-query.clear-password-mapper.enabled=true
+quarkus.security.jdbc.principal-query.clear-password-mapper.password-index=1
+
+quarkus.security.jdbc.principal-query.roles.sql=SELECT r.role_name FROM test_role r, test_user_role ur WHERE ur.username=? AND ur.role_id = r.id
+quarkus.security.jdbc.principal-query.roles.datasource=permissions
+quarkus.security.jdbc.principal-query.roles.attribute-mappings.0.index=1
+quarkus.security.jdbc.principal-query.roles.attribute-mappings.0.to=groups
+--
+
+All the configuration options are detailled bellow.
+

```

Comment:
This table shouldn't be useful. We have automatic doc generation now. The doc explaining how to include it will be published on the website with 0.24.0. I'll point to it.

Reply:
Do you want me to delete the table or not before the merge ?

Reply:
I'll point you to the appropriate doc soon :).

Reply:
Ok I was hoping to have the doc in the 0.24 :/

Reply:
Well, creating the PR the day of the release doesn't help for that :).

Reply:
You can find all the relevant information here: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/extension-authors-guide.adoc#32-generating-the-documentation .

The file will be named `quarkus-elytron-security-jdbc.adoc`.

Note that a major change is coming to what we generate but that won't change how you include it so it should be OK for you.

Reply:
And if you can squash everything when you're done, that will be perfect.